src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/__test__/DiaryDayEntryCellModelTest.swift

```diff
@@ -6,92 +6,611 @@ import Foundation
-		XCTAssertTrue(viewModel.parametersHidden)
+		XCTAssertEqual(cellModel.entryType, .contactPerson)
+		XCTAssertTrue(cellModel.parametersHidden)
 
-		XCTAssertEqual(viewModel.accessibilityTraits, .button)
+		XCTAssertEqual(cellModel.accessibilityTraits, .button)
 	}
 
 	func testContactPersonSelected() throws {
-		let entry: DiaryEntry = .contactPerson(
-			DiaryContactPerson(
-				id: 0,
-				name: "Marcus Scherer",
-				encounter: ContactPersonEncounter(
-					id: 0,
-					date: "2021-02-11",
-					contactPersonId: 0
-				)
-			)
-		)
-		let viewModel = DiaryDayEntryCellModel(entry: entry, dateString: "2021-02-11", store: MockDiaryStore())
+		let cellModel = contactPersonCellModelWithEncounter(name: "Marcus Scherer")
 
-		XCTAssertEqual(viewModel.image, UIImage(named: "Diary_Checkmark_Selected"))
-		XCTAssertEqual(viewModel.text, "Marcus Scherer")
-		XCTAssertEqual(viewModel.font, .enaFont(for: .headline))
+		XCTAssertEqual(cellModel.image, UIImage(named: "Diary_Checkmark_Selected"))
+		XCTAssertEqual(cellModel.text, "Marcus Scherer")
+		XCTAssertEqual(cellModel.font, .enaFont(for: .headline))
 
-		XCTAssertEqual(viewModel.entryType, .contactPerson)
-		XCTAssertFalse(viewModel.parametersHidden)
+		XCTAssertEqual(cellModel.entryType, .contactPerson)
+		XCTAssertFalse(cellModel.parametersHidden)
 
-		XCTAssertEqual(viewModel.accessibilityTraits, [.button, .selected])
+		XCTAssertEqual(cellModel.accessibilityTraits, [.button, .selected])
 	}
 
 	func testLocationUnselected() throws {
-		let entry: DiaryEntry = .location(
-			DiaryLocation(
-				id: 0,
-				name: "Bakery"
+		let cellModel = locationCellModelWithoutVisit(name: "Bakery")
+
+		XCTAssertEqual(cellModel.image, UIImage(named: "Diary_Checkmark_Unselected"))
+		XCTAssertEqual(cellModel.text, "Bakery")
+		XCTAssertEqual(cellModel.font, .enaFont(for: .body))
+
+		XCTAssertEqual(cellModel.entryType, .location)
+		XCTAssertTrue(cellModel.parametersHidden)
+
+		XCTAssertEqual(cellModel.accessibilityTraits, .button)
+	}
+
+	func testLocationSelected() throws {
+		let cellModel = locationCellModelWithVisit(name: "Supermarket")
+
+		XCTAssertEqual(cellModel.image, UIImage(named: "Diary_Checkmark_Selected"))
+		XCTAssertEqual(cellModel.text, "Supermarket")
+		XCTAssertEqual(cellModel.font, .enaFont(for: .headline))
+
+		XCTAssertEqual(cellModel.entryType, .location)
+		XCTAssertFalse(cellModel.parametersHidden)
+
+		XCTAssertEqual(cellModel.accessibilityTraits, [.button, .selected])
+	}
+
+	func testDurationValues() {
+		let cellModel = contactPersonCellModelWithoutEncounter()
+
+		let expectedDurationValues: [DiaryDayEntryCellModel.SegmentedControlValue<ContactPersonEncounter.Duration>] = [
+			DiaryDayEntryCellModel.SegmentedControlValue(
+				title: AppStrings.ContactDiary.Day.Encounter.lessThan15Minutes,
+				value: .lessThan15Minutes
+			),
+			DiaryDayEntryCellModel.SegmentedControlValue(
+				title: AppStrings.ContactDiary.Day.Encounter.moreThan15Minutes,
```

Comment:
What about exactly 15 Minutes?

Reply:
I guess you are expected to either meet for 14.9999999 minutes or less and select `Less than 15 Minutes` or for at least 15.0000001 minutes and select `More than 15 minutes` 🤷🏼‍♂️

Reply:
> (Not sure if the viruses watch is that precise 😅)

True 😅
but seriously the states given do not reflect all possibilities. Can we change it to `moreThanorEqual15Minutes`?

Reply:
The UI only gives the options "less than 15 min" of "more than 15 min", so the options given to the user are accurately represented. I guess that is something you would have to discuss with UX first, why we leave out the "equal 15 minutes" case and don't include it in either of the two brackets. But I hope that shouldn't hinder us from merging this PR for now 😁

Reply:
I would like to see a test case then for this one. Maybe it kills the app and we disclosure all the super-secret information we collect to the public. The tinfoils would hunt us down in such a case 😱